README.md

```diff
@@ -1,22 +1,140 @@
+
+```php
+    public static function handlePreIndex(ZendEvent $event)
+    {
+        /** @var Page $document */
+        $document = $event->getTarget();
+        $params = $event->getParams();
+
+        $params["body"]["page"]["customProperty"] = "something";
+    }
+```
+
+## Available hooks
+
+The following hooks are currently available:
+
+### document.elasticsearch.preIndex
+
+This hook is called after the plugin has extracted all the information from the document to index and before it writes the data to Elasticsearch. You can use this hook to write additional/custom properties to the index.
+
+### asset.elasticsearch.preIndex
+
+This hook is the asset equivalent of "document.elasticsearch.preIndex".
+
+
+## Querying
+
+The pluigin provides a simple query builder to make it easy to extract information from Elasticsearch.
+
+### A simple query example
+
+```php
+        use Byng\Pimcore\Elasticsearch\Query\BoolQuery;
+        use Byng\Pimcore\Elasticsearch\Query\MatchQuery;
+        use Byng\Pimcore\Elasticsearch\Query\Query;
+        use Byng\Pimcore\Elasticsearch\Query\QueryBuilder;
+        use Byng\Pimcore\Elasticsearch\Gateway\PageGateway;
+
+        $boolQuery = new BoolQuery();
+        $boolQuery->addMust(new MatchQuery("_all", "something"));
+        
+        $query = new Query($boolQuery);
+        
+        $queryBuilder = new QueryBuilder();
+        $queryBuilder->setQuery($query);
+        $queryBuilder->setSize(10); // number of results to return
+        
+        $pageGateway = PageGateway::getInstance();
+        $resultSet =  $this->pageGateway->query($queryBuilder);
+```
+
+The following json request will be geenrated form the above code and sent to elasticsearch:
+
+```json
+    {
+        "query": {
+            "bool": {
+                "must": [
+                    {
+                        "match": {
+                            "_all": "something"
+                        }
+                    }
+                ]
+            }
+        },
+        "size": 10
+    }
+```
+
+This will retrieve all documents that contain the word "something".
+
+### Todo
 
-* This plugin requires that the `elasticsearch-mapper-attachments` plugin is installed in 
-Elasticsearch. If you do not have it installed, plugin installation in Pimcore will fail.
-* This plugin automatically creates the indices defined in the configuration file, and also updates
-the mapping of some. Please double-check that this will not negatively impact your Elasticsearch 
-indices. You have been warned - we cannot be held responsible for any loss of data as a result of
-the use of this plugin.
+* Provide more complete query documentation
+* Add support for custom queries, i.e. an array which can be converted to json and posted to elasticsearch without any processing if the plugin doesn't support the whole Elasticsearch DSL for querying.
```

Comment:
Remember that we have `QueryInterface` now, do we need this as a todo? I mean, maybe some documentation on how to do it would be good?